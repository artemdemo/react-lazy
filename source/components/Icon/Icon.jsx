import React from 'react';
import PropTypes from 'prop-types';
import classnames from 'classnames';

const Icon = (props) => {
    const { name, className, style } = props;
    const iconClass = classnames(className, `glyphicon glyphicon-${name}`);
    if (name) {
        return (
            <span
                style={style}
                className={iconClass} />
        );
    }

    return null;
};

Icon.propTypes = {
    name: PropTypes.string,
    className: PropTypes.string,
    style: PropTypes.shape({}),
};

Icon.defaultProps = {
    name: null,
    className: null,
    style: null,
};

export default Icon;
