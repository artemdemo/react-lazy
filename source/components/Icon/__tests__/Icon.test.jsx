import React from 'react';
import renderer from 'react-test-renderer';
import Icon from '../Icon';

describe('<Icon>', () => {
    it('Empty render', () => {
        const tree = renderer.create(
            <Icon />,
        ).toJSON();

        expect(tree).toMatchSnapshot();
    });

    it('with name', () => {
        const tree = renderer.create(
            <Icon name='fire' />,
        ).toJSON();

        expect(tree).toMatchSnapshot();
    });
});
