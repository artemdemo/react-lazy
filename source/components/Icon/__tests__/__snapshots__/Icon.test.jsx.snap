// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Icon> Empty render 1`] = `null`;

exports[`<Icon> with name 1`] = `
<span
  className="glyphicon glyphicon-fire"
  style={null}
/>
`;
